rhel7/fixes/bash/grub2_enable_fips_mode.sh

```diff
@@ -3,32 +3,41 @@
 # include remediation functions library
 . /usr/share/scap-security-guide/remediation_functions
 
-if grep --silent ^PRELINKING /etc/sysconfig/prelink ; then
-        sed -i "s/^PRELINKING.*/PRELINKING=no/g" /etc/sysconfig/prelink
-else
-        echo -e "\n# Set PRELINKING to 'no' per security requirements" >> /etc/sysconfig/prelink
-        echo "PRELINKING=no" >> /etc/sysconfig/prelink
+# if prelink package is installed disable it, else ignore
+if rpm -q prelink 2>&1 > /dev/null; then
+	if grep -q '^PRELINKING=' /etc/sysconfig/prelink ; then
+		sed -i 's/^PRELINKING=.*/# Set PRELINKING to "no" per security requirements\nPRELINKING=no/' /etc/sysconfig/prelink
+	else
+        	echo -e "\n# Set PRELINKING to 'no' per security requirements" >> /etc/sysconfig/prelink
+        	echo "PRELINKING=no" >> /etc/sysconfig/prelink
+	fi
+
+	prelink -u -a
 fi
 
-prelink -u -a
-
 package_command install dracut-fips
 
 dracut -f
 
-if [ -e /sys/firmware/efi ]; then
-	BOOT=`df /boot/efi | tail -1 | awk '{print $1 }'`
+# Correct the form of default kernel command line in  grub
+if grep -q '^GRUB_CMDLINE_LINUX=.*fips=.*"'  /etc/default/grub; then
+	# modify the GRUB command-line if a fips= arg already exists
+	sed -i 's/\(^GRUB_CMDLINE_LINUX=".*\)fips=[^[:space:]]*\(.*"\)/\1 fips=1 \2/'  /etc/default/grub
 else
-	BOOT=`df /boot | tail -1 | awk '{ print $1 }'`
+	# no existing fips=arg is present, append it
+	sed -i 's/\(^GRUB_CMDLINE_LINUX=".*\)"/\1 fips=1"/'  /etc/default/grub
 fi
 
-# Correct the form of default kernel command line in /etc/default/grub
-if ! grep -q "^GRUB_CMDLINE_LINUX=\".*fips=1.*\"" /etc/default/grub;
-then
-  # Append 'fips=1' argument to /etc/default/grub (if not present yet)
-  sed -i "s/\(GRUB_CMDLINE_LINUX=\)\"\(.*\)\"/\1\"\2 fips=1\"/" /etc/default/grub
+# Get the UUID of the device mounted at /boot.
+BOOT_UUID=$(findmnt --noheadings --output uuid --target /boot)
+
+if grep -q '^GRUB_CMDLINE_LINUX=".*boot=.*"'  /etc/default/grub; then
+	# modify the GRUB command-line if a boot= arg already exists
+	sed -i 's/\(^GRUB_CMDLINE_LINUX=".*\)boot=[^[:space:]]*\(.*"\)/\1 boot=UUID='"${BOOT_UUID} \2/" /etc/default/ grub
+else
+	# no existing boot=arg is present, append it
+	sed -i 's/\(^GRUB_CMDLINE_LINUX=".*\)"/\1 boot=UUID='${BOOT_UUID}'"/'  /etc/default/grub
```

Comment:
You shouldn't have the double quote past the `BOOT_UUID`: `... \1 boot=UUID='${BOOT_UUID}'"/'` You substitute text before the closing quote, so `\1 boot=UUID='"${BOOT_UUID}/"` will be just fine.

Reply:
You are right, I confused myself by `\1` not containing the ending double quote. Therefore, please disregard my previous comment.